stdlib/3/os/__init__.pyi

```diff
@@ -285,63 +356,133 @@ def fstat(fd: int) -> stat_result: ...
 def fstatvfs(fd: int) -> statvfs_result: ...  # Unix only
 def fsync(fd: int) -> None: ...
 def ftruncate(fd: int, length: int) -> None: ...  # Unix only
+if sys.version_info >= (3, 5):
+    def get_blocking(fd: int) -> bool: ...  # Unix only
+    def set_blocking(fd: int, blocking: bool) -> None: ...  # Unix only
 def isatty(fd: int) -> bool: ...  # Unix only
+if sys.version_info >= (3, 3):
+    def lockf(fd: int, cmd: int, length: int) -> None: ...  # Unix only
 def lseek(fd: int, pos: int, how: int) -> int: ...
-def open(file: _PathType, flags: int, mode: int = ...) -> int: ...
+if sys.version_info >= (3, 3):
+    def open(file: _PathType, flags: int, mode: Optional[int] = ..., *, dir_fd: Optional[int] = ...) -> int: ...
+else:
+    def open(file: _PathType, flags: int, mode: Optional[int] = ...) -> int: ...
 def openpty() -> Tuple[int, int]: ...  # some flavors of Unix
 def pipe() -> Tuple[int, int]: ...
+if sys.version_info >= (3, 3):
+    def pipe2(flags: int) -> Tuple[int, int]: ...  # some flavors of Unix
+    def posix_fallocate(fd: int, offset: int, length: int) -> None: ...  # Unix only
+    def posix_fadvise(fd: int, offset: int, length: int, advice: int) -> None: ...  # Unix only
+    def pread(fd: int, buffersize: int, offset: int) -> bytes: ...  # Unix only
+    def pwrite(fd: int, string: bytes, offset: int) -> int: ...  # Unix only
 def read(fd: int, n: int) -> bytes: ...
+if sys.version_info >= (3, 3):
+    # sendfile() has a different variation on FreeBSD and Mac OS X
+    def sendfile(out_fd: int, in_fd: int, offset: Optional[int], count: int) -> int: ...  # Unix only
```

Comment:
If I'm reading the code right `count` is also Optional on at least some variations. I think we should give the most permissive type.

You're also missing the arguments headers, trailers, and flags. headers and trailers are Sequences of bytes and flags is an int.

Reply:
I didn't have a Mac on which to test the variant supporting headers, trailers, and flags, so I left it out. But I downloaded a FreeBSD image and added an overloaded variant. When testing, I found that `count` is not Optional on Linux or FreeBSD.